src/components/EventsList/index.tsx

```diff
@@ -1,14 +1,30 @@
 import styled from "@emotion/styled";
+import { useRouter } from "next/router";
+import { useState } from "react";
+
+interface categoryProps {
+  curCategory: number;
+}
 
 export default function EventsList() {
+  const [curCategory, setCurCategory] = useState(1);
+  const categories = ["all", "promotion", "collaboration"];
+
+  const router = useRouter();
+
+  const handleClick = (category: number) => {
+    setCurCategory(category);
+    router.push(`/events?category=${categories[category - 1]}`);
+  };
+
   return (
     <Header>
       <h2>이벤트</h2>
-      <CategoryNav>
+      <CategoryNav curCategory={curCategory}>
         <ul className="events-category-list">
-          <li className="current">전체</li>
-          <li className="events-promotion">프로모션</li>
-          <li className="events-collaboration">콜라보레이션</li>
+          <li onClick={() => handleClick(1)}>전체</li>
+          <li onClick={() => handleClick(2)}>프로모션</li>
+          <li onClick={() => handleClick(3)}>콜라보레이션</li>
```

Comment:
p4 : 현재는 메뉴가 3개지만 몇개 더 추가된다고 하면 배열에 또 넣고 그 배열의 숫자 값을 또 계산하는 절차가 있을거 같아 보입니다. 
차라리 union Type으로 해서 enum으로 거기서 선택한 카테고리 텍스트를 가지고 판단해서 이동시켜주는 편은 더 좋지 않을지 의견드려봅니다. 
```suggestion
          <li onClick={() => handleClick("all")}>전체</li>
          <li onClick={() => handleClick("promotion")}>프로모션</li>
          <li onClick={() => handleClick("collaboration")}>콜라보레이션</li>
```

Reply:
DOM접근 없이 선택된 카테고리를 nth-child로 css를 주려고 이렇게 해봤는데, 좀 더 고민해보겠습니다!

Reply:
아하 😀!! 그럴려고 넘버쓰신거군요! ㅎㅎ  클래스이름을 props로 받아서 그거에 따라서 스타일링일 변화를 줘도 될거 같습니다.